From b5c8bf99d6bac7617c5eea557fafebb2b4e7eb39 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jir=CC=8Ci=CC=81=20Techet?= <techet@gmail.com>
Date: Sat, 21 May 2016 20:16:58 +0200
Subject: [PATCH] Don't use O_NOATIME as it doesn't exist on OS X

---
 src/vteutils.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/vteutils.c b/src/vteutils.c
index 1371cb2..c552ada 100644
--- a/src/vteutils.c
+++ b/src/vteutils.c
@@ -48,7 +48,7 @@ _vte_mkstemp (void)
 
 #ifdef O_TMPFILE
         fd = open (g_get_tmp_dir (),
-                   O_TMPFILE | O_EXCL | O_RDWR | O_NOATIME,
+                   O_TMPFILE | O_EXCL | O_RDWR,
                    0600);
         if (fd != -1)
                 goto done;
@@ -63,8 +63,6 @@ _vte_mkstemp (void)
         unlink (file_name);
         g_free (file_name);
 
-        do { } while (fcntl (fd, F_SETFL, O_NOATIME) == -1 && errno == EINTR);
-
 #ifdef O_TMPFILE
  done:
 #endif
-- 
2.8.1

