From 8e9574627ae0636706185607150197397e02fad6 Mon Sep 17 00:00:00 2001
From: John Ralls <jralls@ceridwen.us>
Date: Fri, 14 Aug 2020 14:16:52 -0700
Subject: [PATCH] Set libfreetype's install name so that meson can link it
 correctly.

---
 CMakeLists.txt | 27 +++++++++++++++------------
 1 file changed, 15 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c9befb4..5395a46 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -426,17 +426,21 @@ target_include_directories(
 )
 
 
-if (BUILD_FRAMEWORK)
-  set_property(SOURCE ${PUBLIC_CONFIG_HEADERS}
-    PROPERTY MACOSX_PACKAGE_LOCATION Headers/config
-  )
-  set_target_properties(freetype PROPERTIES
-    FRAMEWORK TRUE
-    MACOSX_FRAMEWORK_INFO_PLIST builds/mac/freetype-Info.plist
-    PUBLIC_HEADER "${PUBLIC_HEADERS}"
-    XCODE_ATTRIBUTE_INSTALL_PATH "@rpath"
-  )
-endif ()
+if (APPLE)
+  if (BUILD_FRAMEWORK)
+    set_property(SOURCE ${PUBLIC_CONFIG_HEADERS}
+      PROPERTY MACOSX_PACKAGE_LOCATION Headers/config
+      )
+    set_target_properties(freetype PROPERTIES
+      FRAMEWORK TRUE
+      MACOSX_FRAMEWORK_INFO_PLIST builds/mac/freetype-Info.plist
+      PUBLIC_HEADER "${PUBLIC_HEADERS}"
+      XCODE_ATTRIBUTE_INSTALL_PATH "@rpath"
+      )
+  else ()
+    set_target_properties (freetype PROPERTIES MACOSX_RPATH FALSE INSTALL_NAME_DIR "$<INSTALL_PREFIX>/lib")
+  endif ()
+endif()
 
 
 set(PKG_CONFIG_REQUIRED_PRIVATE "")
@@ -469,7 +473,6 @@ if (BROTLIDEC_FOUND)
   list(APPEND PKG_CONFIG_REQUIRED_PRIVATE "libbrotlidec")
 endif ()
 
-
 # Installation
 include(GNUInstallDirs)
 
-- 
2.2.2

