--- a/Makefile.in	2022-12-21 18:44:37.000000000 -0800
+++ b/Makefile.in	2023-06-27 16:14:13.000000000 -0700
@@ -697,7 +697,7 @@
  include/nasmint.h include/nasmlib.h include/nctype.h include/opflags.h \
  include/perfhash.h include/strlist.h include/tables.h include/warnings.h \
  x86/iflaggen.h x86/insnsi.h x86/regs.h
-asm/warnings.$(O): asm/warnings.c config/config.h config/msvc.h \
+asm/warnings.$(O): $(top_srcdir)asm/warnings.c config/config.h config/msvc.h \
  config/unconfig.h config/unknown.h config/watcom.h include/compiler.h \
  include/error.h include/nasmint.h include/warnings.h
 common/common.$(O): common/common.c asm/directiv.h asm/pptok.h asm/preproc.h \

Diff finished.  Tue Jun 27 16:14:20 2023
