--- a/Makefile.in	2022-01-17 06:46:40.000000000 -0800
+++ b/Makefile.in	2023-07-01 14:35:10.000000000 -0700
@@ -2908,7 +2908,7 @@
 	trust/$(DEPDIR)/$(am__dirstamp)
 
 p11-kit-trust.la: $(p11_kit_trust_la_OBJECTS) $(p11_kit_trust_la_DEPENDENCIES) $(EXTRA_p11_kit_trust_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(p11_kit_trust_la_LINK) $(am_p11_kit_trust_la_rpath) $(p11_kit_trust_la_OBJECTS) $(p11_kit_trust_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(p11_kit_trust_la_LINK) $(am_p11_kit_trust_la_rpath) $(p11_kit_trust_la_OBJECTS) $(p11_kit_trust_la_LIBADD) $(LIBS) $(LIBINTL)
 common/frob-getauxval.$(OBJEXT): common/$(am__dirstamp) \
 	common/$(DEPDIR)/$(am__dirstamp)
 

