--- a/Makefile.in	2023-02-19 19:31:46.135333554 +0700
+++ b/Makefile.in	2023-02-19 19:32:28.379010562 +0700
@@ -148,9 +148,9 @@
 SOURCES =
 DIST_SOURCES =
 RECURSIVE_TARGETS = all-recursive check-recursive cscopelist-recursive \
-	ctags-recursive dvi-recursive html-recursive info-recursive \
+	ctags-recursive dvi-recursive info-recursive \
 	install-data-recursive install-dvi-recursive \
-	install-exec-recursive install-html-recursive \
+	install-exec-recursive \
 	install-info-recursive install-pdf-recursive \
 	install-ps-recursive install-recursive installcheck-recursive \
 	installdirs-recursive pdf-recursive ps-recursive \
@@ -400,7 +400,6 @@
 host_cpu = @host_cpu@
 host_os = @host_os@
 host_vendor = @host_vendor@
-htmldir = @htmldir@
 includedir = @includedir@
 infodir = @infodir@
 install_sh = @install_sh@
@@ -908,10 +907,6 @@
 
 dvi-am:
 
-html: html-recursive
-
-html-am:
-
 info: info-recursive
 
 info-am:
@@ -924,10 +919,6 @@
 
 install-exec-am:
 
-install-html: install-html-recursive
-
-install-html-am:
-
 install-info: install-info-recursive
 
 install-info-am:
@@ -972,10 +963,10 @@
 	dist-all dist-bzip2 dist-gzip dist-lzip dist-shar dist-tarZ \
 	dist-xz dist-zip distcheck distclean distclean-generic \
 	distclean-hdr distclean-libtool distclean-tags distcleancheck \
-	distdir distuninstallcheck dvi dvi-am html html-am info \
+	distdir distuninstallcheck dvi dvi-am info \
 	info-am install install-am install-data install-data-am \
 	install-dist_docDATA install-dvi install-dvi-am install-exec \
-	install-exec-am install-html install-html-am install-info \
+	install-exec-am install-info \
 	install-info-am install-man install-pdf install-pdf-am \
 	install-pkgconfigDATA install-ps install-ps-am install-strip \
 	installcheck installcheck-am installdirs installdirs-am \
