--- a/Source/WebKit/UIProcess/gtk/WebDataListSuggestionsDropdownGtk.cpp	2020-02-04 02:24:08.000000000 -0800
+++ b/Source/WebKit/UIProcess/gtk/WebDataListSuggestionsDropdownGtk.cpp	2020-10-20 14:55:56.000000000 -0700
@@ -33,7 +33,7 @@
 #include <WebCore/GtkUtilities.h>
 #include <WebCore/IntPoint.h>
 #include <wtf/glib/GRefPtr.h>
-#include <wtf/glib/GUniquePtr.h>
+#include <WebCore/GUniquePtrGtk.h>
 
 namespace WebKit {
 
