From e2e19c1519dd10f9c64dbcb42cb42fe2326887a1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jir=CC=8Ci=CC=81=20Techet?= <techet@gmail.com>
Date: Mon, 6 Mar 2023 23:22:12 +0100
Subject: [PATCH] Workaround meson expecting git repository presence

---
 meson.build | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/meson.build b/meson.build
index 6944a423b..7b6213870 100644
--- a/meson.build
+++ b/meson.build
@@ -165,13 +165,7 @@ python = pymod.find_installation('python3', modules: doxygen.found() and get_opt
 # These two are truly optional
 rst2html = find_program('rst2html', required: get_option('html-docs'))
 rst2pdf = find_program('rst2pdf', required: get_option('pdf-docs'))
-git = find_program('git', required: false)
-if git.found()
-	ret = run_command(git, 'rev-parse', '--short', '--revs-only', 'HEAD', check: true)
-	cdata.set_quoted('REVISION', ret.stdout().strip())
-else
-	cdata.set_quoted('REVISION', '-1')
-endif
+cdata.set_quoted('REVISION', '-1')
 
 python_command = get_option('python-command')
 if python_command == '' or python_command == 'auto'
-- 
2.37.5

